CMakeLists.txt

```diff
@@ -293,6 +275,58 @@ ELSE ()
 	ENDIF()
 ENDIF()
 
+IF (USE_HTTPS STREQUAL "ON")
+	IF (SECURITY_FOUND)
+		# OS X 10.7 and older do not have some functions we use, fall back to OpenSSL there
+		CHECK_LIBRARY_EXISTS("${SECURITY_DIRS}" SSLCreateContext "Security/SecureTransport.h" HAVE_NEWER_SECURITY)
+		IF (HAVE_NEWER_SECURITY)
+			MESSAGE("-- Found Security ${SECURITY_DIRS}")
+			LIST(APPEND LIBGIT2_PC_LIBS "-framework Security")
+			SET(HTTPS_BACKEND "SecureTransport")
+		ELSE()
+			MESSAGE("-- Security framework is too old, falling back to OpenSSL")
+			SET(SECURITY_FOUND "NO")
+			SET(SECURITY_DIRS "")
+			SET(SECURITY_DIR "")
+			SET(HTTPS_BACKEND "OpenSSL")
+		ENDIF()
+	ELSEIF(WINHTTP)
+		SET(HTTPS_BACKEND "WinHTTP")
+	ELSE()
+		SET(HTTPS_BACKEND "OpenSSL")
+	ENDIF()
+ELSE()
+	SET(HTTPS_BACKEND ${USE_HTTPS})
+ENDIF()
+
+MESSAGE(STATUS "Using HTTPS backend ${HTTPS_BACKEND}")
+
+IF (HTTPS_BACKEND STREQUAL "SecureTransport")
+	IF (NOT SECURITY_FOUND)
+		MESSAGE(FATAL_ERROR "Asked for SecureTransport HTTPS backend, but it wasn't found")
+	ENDIF()
+
+	ADD_DEFINITIONS(-DGIT_SECURE_TRANSPORT)
+	INCLUDE_DIRECTORIES(${SECURITY_INCLUDE_DIR})
+ELSEIF (HTTPS_BACKEND STREQUAL "OpenSSL")
+	IF (NOT OPENSSL_FOUND)
+		MESSAGE(FATAL_ERROR "Asked for OpenSSL HTTPS backend, but it wasn't found")
+	ENDIF()
+
+	ADD_DEFINITIONS(-DGIT_OPENSSL)
+	INCLUDE_DIRECTORIES(${OPENSSL_INCLUDE_DIR})
+	SET(SSL_LIBRARIES ${OPENSSL_LIBRARIES})
+ELSEIF(HTTPS_BACKEND STREQUAL "WinHTTP")
+ENDIF()
+
+IF (USE_HTTPS AND NOT HTTPS_BACKEND)
+	MESSAGE(FATAL_ERROR "Asked for backend " ${HTTPS_BACKEND} " but it wasn't found")
+ENDIF()
+
+IF (HTTPS_BACKEND)
+	ADD_DEFINITIONS(-DGIT_HTTPS)
+ENDIF()
```

Comment:
Seeing all this code we could even think about making this a separate CMake module later on. Not required for this PR, but I'll tackle this at some point, I guess.